internal/env.go

```diff
@@ -25,6 +27,7 @@ func BoolEnv(key string, def bool) bool {
 func IntEnv(key string, def int) int {
 	v, err := strconv.Atoi(os.Getenv(key))
 	if err != nil {
+		log.Warn("Non-integer value for env var %s, defaulting to %d. Parse failed with error: %v", key, def, err)
```

Comment:
Oh, wait. Actually, this will return an error even if the env. var. is not set. So careful here, it could turn into a regression. If the env. var. is not set, we should return a default, otherwise, log an error + the default return.
```go
vv, ok := os.LookupEnv(key)
if !ok {
    return def
}
v, err := strconv.Atoi(vv)
// ...
```